utils/appliance/endpoints/ui.py

```diff
@@ -86,7 +86,7 @@ def pre_navigate(self, _tries=0):
             logger.debug(store.current_appliance.ssh_client.run_command(
                 'top -c -b -n1 -o "%MEM" | head -30').output)  # noqa
             logger.debug('Managed Providers:')
-            logger.debug(store.current_appliance.managed_providers)
+            logger.debug([prov.key for prov in store.current_appliance.managed_providers])
```

Comment:
Yep but you didn't specify what way would you prefer to have it optimized. Would you mind to elaborate? The way I see it, if we want to keep passing full CRUD objects around and not just keys, there would have to be a function to extract keys for situations like this one. Unless you want to introduce a `managed_providers_keys` method...